osu.Server.QueueProcessor/QueueProcessor.cs

```diff
@@ -231,6 +231,22 @@ private void outputStats()
 
         public void ClearQueue() => redis.GetDatabase().KeyDelete(QueueName);
 
+        /// <summary>
+        /// Publishes a message to a Redis channel with the supplied <paramref name="channelName"/>.
+        /// </summary>
+        /// <remarks>
+        /// The message will be serialised using JSON.
+        /// Successful publications are tracked in Datadog, using the <paramref name="channelName"/> and the <typeparamref name="TMessage"/>'s full type name as a tag.
+        /// </remarks>
+        /// <param name="channelName">The name of the Redis channel to publish to.</param>
+        /// <param name="message">The message to publish to the channel.</param>
+        /// <typeparam name="TMessage">The type of message to be published.</typeparam>
+        public void PublishMessage<TMessage>(string channelName, TMessage message)
+        {
+            redis.GetDatabase().Publish(channelName, JsonConvert.SerializeObject(message));
+            DogStatsd.Increment("messages_published", tags: new[] { $"channel:{channelName}", $"type:{typeof(TMessage).FullName}" });
```

Comment:
Dunno if this is overkill or not. Given that Redis's pubsub is a no-guarantees implementation, it felt like it may be nice to get counts from the publisher directly? I don't see datadog having anything in-house that could add this sort of tracking. Definitely think it's a good idea to have some sort of observability at a few stages of this flow to be able to diagnose failures better, though.